src/store/models/taro.ts

```diff
@@ -54,6 +74,64 @@ const taroModel: TaroModel = {
     const balances = await api.listBalances(node);
     actions.setBalances({ node, balances });
   }),
+  getAllInfo: thunk(async (actions, node) => {
+    await actions.getAssets(node);
+    await actions.getBalances(node);
+  }),
+
+  mintAsset: thunk(
+    async (actions, payload, { injections, getStoreState, getStoreActions }) => {
+      const {
+        node,
+        assetType,
+        name,
+        amount,
+        metaData,
+        enableEmission,
+        skipBatch,
+        autoFund,
+      } = payload;
+      const TaroNode = node as TarodNode;
+      const network = getStoreState().network.networkById(TaroNode.networkId);
+      const lndNode = network.nodes.lightning.find(
+        n => n.name === TaroNode.lndName,
+      ) as LightningNode;
+      //fund lnd node
+      if (autoFund) {
+        await getStoreActions().lightning.depositFunds({ node: lndNode, sats: '10000' });
+      }
+
+      //mint taro asset
+      const taroapi = injections.taroFactory.getService(node);
+      const req: TARO.MintAssetRequest = {
+        assetType: assetType ? TARO_ASSET_TYPE.COLLECTIBLE : TARO_ASSET_TYPE.NORMAL,
+        name,
+        amount: assetType ? 1 : amount,
+        metaData: Buffer.from(metaData, 'hex'),
+        enableEmission,
+        skipBatch,
+      };
+      const res = await taroapi.mintAsset(node, req);
+      if (res.batchKey) {
+        actions.setBatchKey({ node, newBatchKey: res.batchKey.toString('hex') });
+      }
+      //update network
+      const btcNode =
+        network.nodes.bitcoin.find(n => n.name === lndNode.backendName) ||
+        network.nodes.bitcoin[0];
+      await getStoreActions().bitcoind.mine({
+        blocks: BLOCKS_TIL_CONFIRMED,
+        node: btcNode,
+      });
+      await getStoreActions().lightning.waitForNodes([lndNode]);
+      await getStoreActions().designer.syncChart(network);
```

Comment:
Any reason why we need to sync the chart after minting an asset? That just updates the channels and backend connections in the designer which minting shouldn't affect.